config-model-generator/src/main/java/io/strimzi/build/kafka/metadata/CommonConfigModelGenerator.java

```diff
@@ -0,0 +1,202 @@
+        } else if ("int".equals(typeStr)) {
+            type = Type.INT;
+        } else if ("short".equals(typeStr)) {
+            type = Type.SHORT;
+        } else if ("long".equals(typeStr)) {
+            type = Type.LONG;
+        } else if ("double".equals(typeStr)) {
+            type = Type.DOUBLE;
+        } else if ("list".equals(typeStr)) {
+            type = Type.LIST;
+        } else {
+            throw new RuntimeException("Unsupported type: " + typeStr);
+        }
+        return type;
+    }
+
+    protected Scope parseScope(String scopeStr) {
+        Scope scope;
+        if ("per-broker".equals(scopeStr)) {
+            scope = Scope.PER_BROKER;
+        } else if ("cluster-wide".equals(scopeStr)) {
+            scope = Scope.CLUSTER_WIDE;
+        } else if ("read-only".equals(scopeStr)) {
+            scope = Scope.READ_ONLY;
+        } else {
+            throw new RuntimeException("Unsupported scope: " + scopeStr);
+        }
+        return scope;
+    }
+
+    protected List<String> validList(ConfigDef.ConfigKey key) {
+        try {
+            Field f = ConfigDef.ValidList.class.getDeclaredField("validString");
+            PrivilegedAction<Object> pa = () -> {
+                f.setAccessible(true);
+                return null;
+            };
+            AccessController.doPrivileged(pa);
+            ConfigDef.ValidString itemValidator = (ConfigDef.ValidString) f.get(key.validator);
+            List<String> validItems = enumer(itemValidator);
+            return validItems;
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    protected List<String> enumer(ConfigDef.Validator validator) {
+        try {
+            Field f = getField(ConfigDef.ValidString.class, "validStrings");
+            return (List<String>) f.get(validator);
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    protected Field getOneOfFields(Class<?> cls, String... alternativeFields) {
+        for (String field : alternativeFields)  {
+            try {
+                return getField(cls, field);
+            } catch (RuntimeException e)    {
+                if (e.getCause() instanceof NoSuchFieldException)   {
+                    continue;
+                } else {
+                    throw e;
+                }
+            }
+        }
+        throw new RuntimeException("None of the alternative fields were found.");
+    }
+
+    protected Field getField(Class<?> cls, String fieldName) {
+        return AccessController.doPrivileged((PrivilegedAction<Field>) () -> {
+            try {
+                Field f1 = cls.getDeclaredField(fieldName);
+                f1.setAccessible(true);
+                return f1;
+            } catch (ReflectiveOperationException e) {
+                throw new RuntimeException(e);
+            }
```

Comment:
It would be cleaner to catch the `NoSuchFieldException` here and return `null`, rather than catching `NoSuchFieldException` in `getOneOfFields`. (I'm not really a fan of using exceptions to indicate something that's not an error condition, but could be reported via the method result).